cognitive/src/main/scala/com/microsoft/azure/synapse/ml/services/GlobalParam.scala

```diff
@@ -0,0 +1,213 @@
+package com.microsoft.azure.synapse.ml.services
+
+import com.microsoft.azure.synapse.ml.param.ServiceParam
+import org.apache.spark.ml.param.Param
+import org.apache.spark.sql.Row
+
+import scala.collection.mutable
+
+sealed trait GlobalKey[T]
+sealed trait GlobalParamKey[T] extends GlobalKey[T]
+sealed trait GlobalServiceParamKey[T] extends GlobalKey[Either[T, String]] //left is Scalar, right is Vector
+
+case object OpenAIDeploymentNameKey extends GlobalServiceParamKey[String]
+
+object GlobalParams {
+  private val ParamToKeyMap: mutable.Map[Any, GlobalKey[_]] = mutable.Map.empty
+  private val GlobalParams: mutable.Map[GlobalParamKey[Any], Any] = mutable.Map.empty
+  private val GlobalServiceParams: mutable.Map[GlobalServiceParamKey[Any], Either[Any, String]] = mutable.Map.empty
```

Comment:
it seems odd that we need to special case service params here, could this be one map for all param types?

Reply:
That's how I had it originally - but because I needed separate getters for the ServiceParams, I flexed them out to enforce the types. 

The getters are: 
def getServiceParamScalar[T](sp: ServiceParam[T]): Option[T] = {
def getServiceParamVector[T](sp: ServiceParam[T]): Option[String] = {

I added GlobalServiceParams to explicitly have a dictionary that maps to a `Either[T, String]` so that we can enforce types on ServiceParams. With just one dictionary it'd have to map to `Any`, meaning I'd be blindly checking if it's an Either type or not. 

So I guess it could work with just one dictionary, but the typing would just be looser.

Reply:
Brought it down to only one Key and Dictionary. However, added a boolean in the Key initialization and asserts to ensure that Param functions are used for Params and ServiceParam functions are used for Service Params.